core/src/test/java/com/backblaze/b2/client/B2StorageClientWebifierImplTest.java

```diff
@@ -1190,7 +1191,9 @@ public void testDownloadById() throws B2Exception {
         final B2DownloadByIdRequest request = B2DownloadByIdRequest
                 .builder(fileId(1))
                 .build();
-        webifier.downloadById(ACCOUNT_AUTH, request, noopContentHandler);
+        final B2StorageClientWebifierImpl webifierSpy = spy(webifier);
+        webifierSpy.downloadById(ACCOUNT_AUTH, request, noopContentHandler);
+        verify(webifierSpy).getDownloadByIdUrl(ACCOUNT_AUTH, request);
```

Comment:
hmm... I think this tests that getDownloadByIdUrl is _called_ but my brain is wondering how useful this verify actually is beyond that. isn't that just an implementation detail? how does verifying it actually tell us more about whether downloadById is working as designed?

Reply:
I was thinking that the combination of the verify and the assertion in the test would test that `downloadById()` is working as designed.

The verify checks that the getDownloadByIdUrl() is being called, which is an implementation detail, but an important one as that was the PR change

This verify alone doesn't test that the method is working properly however along with the assertion that the value from the verify call  (expectedUrl) is the same as the one being passed to the web client call, we test that we call the right thing and that the right thing is passed.